### PR TITLE
fix(schedule): Fix for swabbie scheduling

### PR DESCRIPTION
**Pull request by jeyrschabu**
- remove double scheduling after caches are loaded
- refactored ScheduledAgent#timeToWork to not include off hours
- if off hours are needed, the config can be explicitly updated

### CODE REVIEW

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt [hunk 1]
```diff
@@ -25,12 +25,6 @@ import java.time.LocalDateTime
 import java.time.LocalTime
 import java.time.ZoneId
 
-/**
```
Comment (emjburns) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt]: Why did you delete this comment? I think it's helpful.

Comment (jeyrschabu) [reply swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt]: I was feeling like deleting some comments I guess. Will put them back.

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt [hunk 2]
```diff
@@ -75,59 +76,49 @@ abstract class ScheduledAgent(
...
-      try {
-        while (!cacheStatus.cachesLoaded()) {
-          log.info("Can't work until the caches load...")
-          Thread.sleep(Duration.ofSeconds(5).toMillis())
-        }
-        log.info("Caches loaded.")
-        scheduleSwabbie()
-      } catch (e: Exception) {
-        log.error("Failed while waiting for cache to start in ${javaClass.simpleName}.", e)
+    try {
```
Comment (emjburns) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt]: Great catch :)

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt (hunk 1 above)
Comment (emjburns) [reply swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt]: Thank you!

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt [hunk 3]
```diff
@@ -75,59 +76,49 @@ abstract class ScheduledAgent(
...
-                timeToWork(swabbieProperties.schedule, clock) -> {
-                  if (dynamicConfigService.isEnabled("swabbie.work", true)) {
-                    log.info("Swabbie schedule: working")
-                    runSwabbie()
-                  } else {
-                    log.info("Swabbie schedule: disabled via property swabbie.work")
-                  }
+    runningTask = executorService.scheduleWithFixedDelay({
+      when {
+        dynamicConfigService.isEnabled(SWABBIE_FLAG_PROPERY, true) -> {
```
Comment (emjburns) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt]: This condition seems to be `if swabbie.work == true then do not work`. Based on the log message, this seems to be flipped

Comment (jeyrschabu) [reply swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt]: Yeah good catch. I caught that in the other PR I was working last night. Will update

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt [hunk 4]
```diff
@@ -168,21 +159,14 @@ abstract class ScheduledAgent(
...
-      val now = LocalTime.from(clock.instant().atZone(schedule.getZoneId()))
-
-      return if (startTime <= endTime) {
-        // Work during the day
-        now.isAfter(startTime) && now.isBefore(endTime)
-      } else {
-        // Work during the night
-        now.isAfter(startTime) || now.isBefore(endTime)
-      }
+      return if (schedule.enabled && LocalDate.now(clock).dayOfWeek in schedule.allowedDaysOfWeek) {
```
Comment (emjburns) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt]: I personally don't like lifting the return out of the if statement because I feel like it pulls us closer to the groovy "oh the last line is implicitly the return of the function, and I find that hard to easily parse when reading code. I prefer to keep the two returns.

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt [hunk 5]
```diff
@@ -168,21 +159,14 @@ abstract class ScheduledAgent(
...
+        val startTime: LocalTime = schedule.getResolvedStartTime()
+        val endTime: LocalTime = schedule.getResolvedEndTime()
+        val now = LocalTime.from(clock.instant().atZone(schedule.getZoneId()))
+
+        startTime.isBefore(endTime) &&
```
Comment (emjburns) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt]: If it's no longer valid to have a start time after your end time (ex. working from 8 pm to 8 am) then we should throw an exception or print some error message

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt (hunk 4 above)
Comment (jeyrschabu) [reply swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt]: Think it's different than the groovy last line is the return thing. But for readability, I can see the traditional return idiom to be clearer.

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt [hunk 6]
```diff
@@ -26,9 +26,9 @@ import java.time.LocalTime
 import java.time.ZoneId
 
 /**
- * @param minImagesUsedByLC minimum number of images used by launch configurations that should be
+ * @param {@link #minImagesUsedByLC} minimum number of images used by launch configurations that should be
```
Comment (jeyrschabu) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt]: quick fix to the doc here
